scipy/spatial/_hausdorff.pyx

```diff
@@ -0,0 +1,63 @@
+"""
+Directed Hausdorff Code
+
+.. versionadded:: 0.19.0
+
+"""
+#
+# Copyright (C)  Tyler Reddy and Richard Gowers, 2016
+#
+# Distributed under the same BSD license as Scipy.
+#
+
+import numpy as np
+cimport numpy as np
+cimport cython
+from libc.math cimport sqrt
+
+__all__ = ['directed_hausdorff']
+
+@cython.boundscheck(False)
+def directed_hausdorff(double[:,::1] ar1, double[:,::1] ar2):
+
+    cdef double cmax, cmin, d
+    cdef int break_occurred
+    cdef int N1 = ar1.shape[0]
+    cdef int N2 = ar2.shape[0]
+    cdef int data_dims = ar1.shape[1]
+    cdef unsigned int i, j, k, i_store, j_store, i_ret, j_ret
+    cdef long[:] resort1, resort2
+
+    # shuffling the points in each array generally increases the likelihood of
+    # an advantageous break in the inner search loop and never decreases the
+    # performance of the algorithm
+    np.random.seed(123)
+    np.random.shuffle(np.asarray(ar1))
+    np.random.shuffle(np.asarray(ar2))
+    resort1 = np.arange(N1)
+    resort2 = np.arange(N2)
+    np.random.shuffle(np.asarray(resort1))
+    np.random.shuffle(np.asarray(resort2))
```

Comment:
I'm not sure how this can work. How are you sure that `ar1` and `resort1` are shuffled the same was? My guess is that you have to reseed before calling shuffle again.

Wouldn't be better to first create an indexing array and shuffle that. Then we can use it to shuffle `ar1` and `ar2`.
This also avoid the need to set a specific seed. 

```python
resort1 = np.arange(N1)
resort2 = np.arange(N2)
np.random.shuffle(resort1)
np.random.shuffle(resort2)
ar1 = ar1[resort1]
ar2 = ar2[resort2]
```